### PR TITLE
Add apps pipeline

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward giantswarm/giantswarm#16664

Using the last sonobuoy update https://github.com/giantswarm/sonobuoy-plugin/pull/44 , I would like to add `apps` pipeline.

### CODE REVIEW

File: tekton/tasks/apps.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      script: |
+        #! /bin/sh
+        /sonobuoy run \
+            --kubeconfig "${KUBECONFIG_PATH}" \
+            --namespace "$(cat "${CLUSTER_ID_PATH}")-sonobuoy" \
+            --plugin $(workspaces.cluster.path)/giantswarm-plugin.yaml \
+            --plugin-env giantswarm.CP_KUBECONFIG="$(cat "${KUBECONFIG_PATH}")" \
+            --plugin-env giantswarm.CLUSTER_ID="$(cat "${CLUSTER_ID_PATH}")" \
+            --plugin-env giantswarm.E2E_FOCUS="App" \
+            --mode=quick \
```
Comment (tomahawk28) [inline tekton/tasks/apps.yaml:36]: The only difference between `azure` and `apps` is this, we would run the `apps_test.go` exclusively.

File: tekton/tasks/apps.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  workspaces:
+    - name: cluster
+      description: Cluster information is stored here.
+  params:
+    - name: kubeconfig-path
+      type: string
+      description: Management Cluster kubeconfig path.
+  steps:
+    - name: download-sonobuoy-plugin
+      image: busybox
```
Comment (MarcelMue) [inline tekton/tasks/apps.yaml:16]: We should change this image - not sure if this is the only place we use it though

File: tekton/tasks/apps.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          value: $(params.kubeconfig-path)
+        - name: CLUSTER_ID_PATH
+          value: $(workspaces.cluster.path)/cluster-id
+      script: |
+        #! /bin/sh
+        /sonobuoy run \
+            --kubeconfig "${KUBECONFIG_PATH}" \
+            --namespace "$(cat "${CLUSTER_ID_PATH}")-sonobuoy" \
+            --plugin $(workspaces.cluster.path)/giantswarm-plugin.yaml \
+            --plugin-env giantswarm.CP_KUBECONFIG="$(cat "${KUBECONFIG_PATH}")" \
```
Comment (MarcelMue) [inline tekton/tasks/apps.yaml:33]: Is the management cluster kubeconfig actually needed?

Comment (tomahawk28) [reply tekton/tasks/apps.yaml:33]: Yes, we only need a CP kubeconfig for the apps test to check the cluster namespace.

File: tekton/tasks/apps.yaml (hunk 2 above)
Comment (tomahawk28) [reply tekton/tasks/apps.yaml:16]: @MarcelMue We have used the busybox image on `apps` and `azure` tests.  Which image should we use?

Comment (MarcelMue) [reply tekton/tasks/apps.yaml:16]: Let's do this separately then.

**Review by MarcelMue — APPROVED** [review]
Fine with me to try this.

**Review by rossf7 — APPROVED** [review]

**Pull request merged**